test/int.jl

```diff
@@ -266,6 +266,14 @@ end
     U in [Base.BitInteger_types..., BigInt]
     @test typeof(rand(U(0):U(127)) % T) === T
 end
+@testset "x::Signed % Unsigned returns a T, T = $T" for T in [Base.BitSigned_types...]
+    @test typeof(rand(T) % Unsigned) <: Unsigned
```

Comment:
I'd rather this test a few interesting cases instead (or in addition). For example `typemax`, `typemin`, `-typemax`. Also should test for `x` a BigInt, since that's an interesting one for which there is no unsigned type.

Reply:
What do you think? Should `BigInt(x) % Unsigned` always throw an error, or should it handle values that only require 128 signed bits and throw and error for larger values?

Reply:
I was surprised that `unsigned(Int64) == UInt64` but `signed(UInt64)` errors.  We should remedy that, defining `signed(x::T) where {T<:Base.BitUnsigned_types}` to be the size corresponding `BitSigned_types` in multinverses.jl (where the `unsigned(type{T})` are defined.

Reply:
Why support `x::BigInt % Unsigned`? It doesn't really make sense, because `BigInt` is not a bitstype and these conversions are from one n-byte integer type (signed or unsigned) to another n-byte integer type (unsigned or signed).
The error that happens `.. no method matching Unsigned(::BigInt)` seems appropriate,

Reply:
Fixed-size integer types represent modular classes of integer values. When you do `x % T` you are saying "give me the element of that represents the modular class of `x` in `T`. That is a perfectly well-defined operation for `x::BigInt % Unsigned`. In fact, this operation is _always_ well-defined because `x` always represents a specific integer value and that value always belongs to the modular class represented by a unique element of `T`.

Reply:
But in `x::BigInt % Unsigned`, how to choose the destination type? We won't define an `UnsignedBigInt` type. And we can't return `x` because it's expected that `(x % Unsigned) isa Unsigned`.

Reply:
Oh, I see what you mean—I interpreted that as being for some specific `T <: Unsigned` but the issue is for the literal type `Unsigned`. That is quite unclear so agree about it being an error. The alternative would be to choose `UInt` as the "default unsigned type" but that's questionable.

Reply:
> The alternative would be to choose UInt as the "default unsigned type" but that's questionable.

I guess it's always possible to add it later if there is demand for that, but while this makes some sense, this seems a bit weird to return a smaller type than `Int128(1) % Unsigned`.

Reply:
Agree. Which leads to an issue if/when we ever add `(U)Int256` types.